src/main/java/it/gov/pagopa/idpay/transactions/service/RewardTransactionServiceImpl.java

```diff
@@ -49,4 +53,14 @@ public Mono<RewardTransaction> save(Message<String> message) {
     private RewardTransactionDTO deserializeMessage(Message<String> message) {
         return Utils.deserializeMessage(message, objectReader, e -> errorNotifierService.notifyTransaction(message, "Unexpected JSON", true, e));
     }
+
+    @Override
+    public Flux<RewardTransaction> findTrxsFilters(String idTrxAcquirer, String userId, String trxDate, String amount) {
+        return Flux.defer(() ->
```

Comment:
why Flux.defer? I think you could remove this line and just call the findByFilters